### ISSUE TITLE
GracefulShutdownTest.shutdownMemberAndCluster_concurrently_withoutPartitionInitialization

### ISSUE THREAD

**Issue by hasancelik** [hazelcast/hazelcast#8709]
```
java.util.concurrent.ExecutionException: com.hazelcast.core.OperationTimeoutException: ChangeClusterStateOperation invocation failed to complete due to operation-heartbeat-timeout. Current time: 2016-08-12 05:18:34.300. Total elapsed time: 120544 ms. Last operation heartbeat: never. Last operation heartbeat from member: never. Invocation{op=com.hazelcast.internal.cluster.impl.operations.ChangeClusterStateOperation{serviceName='hz:core:clusterService', identityHash=1167652188, partitionId=-1, replicaIndex=0, callId=0, invocationTime=1470968193738 (2016-08-12 05:16:33.738), waitTimeout=-1, callTimeout=60000}, tryCount=250, tryPauseMillis=500, invokeCount=1, callTimeoutMillis=60000, firstInvocationTimeMs=1470968193756, firstInvocationTime='2016-08-12 05:16:33.756', lastHeartbeatMillis=0, lastHeartbeatTime='1970-01-01 02:00:00.000', target=[127.0.0.1]:8850, pendingResponse={VOID}, backupsAcksExpected=0, backupsAcksReceived=0, connection=MockConnection{localEndpoint=[127.0.0.1]:8851, remoteEndpoint=[127.0.0.1]:8850}}
    at java.util.concurrent.FutureTask.report(FutureTask.java:122)
    at java.util.concurrent.FutureTask.get(FutureTask.java:188)
    at com.hazelcast.internal.partition.GracefulShutdownTest.shutdownMemberAndCluster_concurrently(GracefulShutdownTest.java:465)
    at com.hazelcast.internal.partition.GracefulShutdownTest.shutdownMemberAndCluster_concurrently_withoutPartitionInitialization(GracefulShutdownTest.java:430)
Caused by: com.hazelcast.core.OperationTimeoutException: ChangeClusterStateOperation invocation failed to complete due to operation-heartbeat-timeout. Current time: 2016-08-12 05:18:34.300. Total elapsed time: 120544 ms. Last operation heartbeat: never. Last operation heartbeat from member: never. Invocation{op=com.hazelcast.internal.cluster.impl.operations.ChangeClusterStateOperation{serviceName='hz:core:clusterService', identityHash=1167652188, partitionId=-1, replicaIndex=0, callId=0, invocationTime=1470968193738 (2016-08-12 05:16:33.738), waitTimeout=-1, callTimeout=60000}, tryCount=250, tryPauseMillis=500, invokeCount=1, callTimeoutMillis=60000, firstInvocationTimeMs=1470968193756, firstInvocationTime='2016-08-12 05:16:33.756', lastHeartbeatMillis=0, lastHeartbeatTime='1970-01-01 02:00:00.000', target=[127.0.0.1]:8850, pendingResponse={VOID}, backupsAcksExpected=0, backupsAcksReceived=0, connection=MockConnection{localEndpoint=[127.0.0.1]:8851, remoteEndpoint=[127.0.0.1]:8850}}
    at com.hazelcast.spi.impl.operationservice.impl.InvocationFuture.newOperationTimeoutException(InvocationFuture.java:150)
    at com.hazelcast.spi.impl.operationservice.impl.InvocationFuture.resolve(InvocationFuture.java:98)
    at com.hazelcast.spi.impl.operationservice.impl.InvocationFuture.resolveAndThrow(InvocationFuture.java:74)
    at com.hazelcast.spi.impl.AbstractInvocationFuture.get(AbstractInvocationFuture.java:186)
    at com.hazelcast.util.FutureUtil.executeWithDeadline(FutureUtil.java:326)
    at com.hazelcast.util.FutureUtil.waitWithDeadline(FutureUtil.java:310)
    at com.hazelcast.util.FutureUtil.waitWithDeadline(FutureUtil.java:284)
    at com.hazelcast.transaction.impl.TransactionImpl.commit(TransactionImpl.java:292)
    at com.hazelcast.internal.cluster.impl.ClusterStateManager.changeClusterState(ClusterStateManager.java:274)
    at com.hazelcast.internal.cluster.impl.ClusterStateManager.changeClusterState(ClusterStateManager.java:241)
    at com.hazelcast.internal.cluster.impl.ClusterServiceImpl.changeClusterState(ClusterServiceImpl.java:850)
    at com.hazelcast.internal.cluster.impl.AdvancedClusterStateTest.changeClusterStateEventually(AdvancedClusterStateTest.java:794)
    at com.hazelcast.internal.partition.GracefulShutdownTest$6.run(GracefulShutdownTest.java:459)
    at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:471)
    at java.util.concurrent.FutureTask.run(FutureTask.java:262)
    at java.lang.Thread.run(Thread.java:745)
```

https://hazelcast-l337.ci.cloudbees.com/view/Official%20Builds/job/Hazelcast-3.x-OracleJDK1.7/com.hazelcast$hazelcast/572/testReport/junit/com.hazelcast.internal.partition/GracefulShutdownTest/shutdownMemberAndCluster_concurrently_withoutPartitionInitialization/

**Comment by hasancelik** [comment]
```
java.util.concurrent.ExecutionException: com.hazelcast.core.OperationTimeoutException: ChangeClusterStateOperation invocation failed to complete due to operation-heartbeat-timeout. Current time: 2016-09-19 23:51:43.946. Total elapsed time: 120291 ms. Last operation heartbeat: never. Last operation heartbeat from member: never. Invocation{op=com.hazelcast.internal.cluster.impl.operations.ChangeClusterStateOperation{serviceName='hz:core:clusterService', identityHash=548584131, partitionId=-1, replicaIndex=0, callId=0, invocationTime=1474318183655 (2016-09-19 23:49:43.655), waitTimeout=-1, callTimeout=60000}, tryCount=250, tryPauseMillis=500, invokeCount=1, callTimeoutMillis=60000, firstInvocationTimeMs=1474318183655, firstInvocationTime='2016-09-19 23:49:43.655', lastHeartbeatMillis=0, lastHeartbeatTime='1970-01-01 02:00:00.000', target=[127.0.0.1]:12923, pendingResponse={VOID}, backupsAcksExpected=0, backupsAcksReceived=0, connection=null}
	at java.util.concurrent.FutureTask.report(FutureTask.java:122)
	at java.util.concurrent.FutureTask.get(FutureTask.java:188)
	at com.hazelcast.internal.partition.GracefulShutdownTest.shutdownMemberAndCluster_concurrently(GracefulShutdownTest.java:465)
	at com.hazelcast.internal.partition.GracefulShutdownTest.shutdownMemberAndCluster_concurrently_withoutPartitionInitialization(GracefulShutdownTest.java:430)
Caused by: com.hazelcast.core.OperationTimeoutException: ChangeClusterStateOperation invocation failed to complete due to operation-heartbeat-timeout. Current time: 2016-09-19 23:51:43.946. Total elapsed time: 120291 ms. Last operation heartbeat: never. Last operation heartbeat from member: never. Invocation{op=com.hazelcast.internal.cluster.impl.operations.ChangeClusterStateOperation{serviceName='hz:core:clusterService', identityHash=548584131, partitionId=-1, replicaIndex=0, callId=0, invocationTime=1474318183655 (2016-09-19 23:49:43.655), waitTimeout=-1, callTimeout=60000}, tryCount=250, tryPauseMillis=500, invokeCount=1, callTimeoutMillis=60000, firstInvocationTimeMs=1474318183655, firstInvocationTime='2016-09-19 23:49:43.655', lastHeartbeatMillis=0, lastHeartbeatTime='1970-01-01 02:00:00.000', target=[127.0.0.1]:12923, pendingResponse={VOID}, backupsAcksExpected=0, backupsAcksReceived=0, connection=null}
	at com.hazelcast.spi.impl.operationservice.impl.InvocationFuture.newOperationTimeoutException(InvocationFuture.java:150)
	at com.hazelcast.spi.impl.operationservice.impl.InvocationFuture.resolve(InvocationFuture.java:98)
	at com.hazelcast.spi.impl.operationservice.impl.InvocationFuture.resolveAndThrow(InvocationFuture.java:74)
	at com.hazelcast.spi.impl.AbstractInvocationFuture.get(AbstractInvocationFuture.java:186)
	at com.hazelcast.util.FutureUtil.executeWithDeadline(FutureUtil.java:326)
	at com.hazelcast.util.FutureUtil.waitWithDeadline(FutureUtil.java:310)
	at com.hazelcast.util.FutureUtil.waitWithDeadline(FutureUtil.java:284)
	at com.hazelcast.transaction.impl.TransactionImpl.commit(TransactionImpl.java:292)
	at com.hazelcast.internal.cluster.impl.ClusterStateManager.changeClusterState(ClusterStateManager.java:274)
	at com.hazelcast.internal.cluster.impl.ClusterStateManager.changeClusterState(ClusterStateManager.java:241)
	at com.hazelcast.internal.cluster.impl.ClusterServiceImpl.changeClusterState(ClusterServiceImpl.java:850)
	at com.hazelcast.internal.cluster.impl.AdvancedClusterStateTest.changeClusterStateEventually(AdvancedClusterStateTest.java:794)
	at com.hazelcast.internal.partition.GracefulShutdownTest$6.run(GracefulShutdownTest.java:459)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:471)
	at java.util.concurrent.FutureTask.run(FutureTask.java:262)
	at java.lang.Thread.run(Thread.java:745)
```

https://hazelcast-l337.ci.cloudbees.com/view/Official%20Builds/job/Hazelcast-3.maintenance-ZuluJDK7/com.hazelcast$hazelcast/124/testReport/junit/com.hazelcast.internal.partition/GracefulShutdownTest/shutdownMemberAndCluster_concurrently_withoutPartitionInitialization/

**Comment by hasancelik** [comment]
https://hazelcast-l337.ci.cloudbees.com/view/Hazelcast/job/Hazelcast-3.x-IbmJDK1.8/368/

**Comment by hasancelik** [comment]
```
java.util.concurrent.ExecutionException: com.hazelcast.core.OperationTimeoutException: ChangeClusterStateOperation invocation failed to complete due to operation-heartbeat-timeout. Current time: 2016-10-27 05:47:25.839. Start time: 2016-10-27 05:45:24.965. Total elapsed time: 120874 ms. Last operation heartbeat: never. Last operation heartbeat from member: never. Invocation{op=com.hazelcast.internal.cluster.impl.operations.ChangeClusterStateOperation{serviceName='hz:core:clusterService', identityHash=501998777, partitionId=-1, replicaIndex=0, callId=0, invocationTime=1477536324965 (2016-10-27 05:45:24.965), waitTimeout=-1, callTimeout=60000}, tryCount=250, tryPauseMillis=500, invokeCount=1, callTimeoutMillis=60000, firstInvocationTimeMs=1477536324965, firstInvocationTime='2016-10-27 05:45:24.965', lastHeartbeatMillis=0, lastHeartbeatTime='1970-01-01 02:00:00.000', target=[127.0.0.1]:12202, pendingResponse={VOID}, backupsAcksExpected=0, backupsAcksReceived=0, connection=null}
	at java.util.concurrent.FutureTask.report(FutureTask.java:122)
	at java.util.concurrent.FutureTask.get(FutureTask.java:188)
	at com.hazelcast.internal.partition.GracefulShutdownTest.shutdownMemberAndCluster_concurrently(GracefulShutdownTest.java:465)
	at com.hazelcast.internal.partition.GracefulShutdownTest.shutdownMemberAndCluster_concurrently_withoutPartitionInitialization(GracefulShutdownTest.java:430)
Caused by: com.hazelcast.core.OperationTimeoutException: ChangeClusterStateOperation invocation failed to complete due to operation-heartbeat-timeout. Current time: 2016-10-27 05:47:25.839. Start time: 2016-10-27 05:45:24.965. Total elapsed time: 120874 ms. Last operation heartbeat: never. Last operation heartbeat from member: never. Invocation{op=com.hazelcast.internal.cluster.impl.operations.ChangeClusterStateOperation{serviceName='hz:core:clusterService', identityHash=501998777, partitionId=-1, replicaIndex=0, callId=0, invocationTime=1477536324965 (2016-10-27 05:45:24.965), waitTimeout=-1, callTimeout=60000}, tryCount=250, tryPauseMillis=500, invokeCount=1, callTimeoutMillis=60000, firstInvocationTimeMs=1477536324965, firstInvocationTime='2016-10-27 05:45:24.965', lastHeartbeatMillis=0, lastHeartbeatTime='1970-01-01 02:00:00.000', target=[127.0.0.1]:12202, pendingResponse={VOID}, backupsAcksExpected=0, backupsAcksReceived=0, connection=null}
	at com.hazelcast.spi.impl.operationservice.impl.InvocationFuture.newOperationTimeoutException(InvocationFuture.java:151)
	at com.hazelcast.spi.impl.operationservice.impl.InvocationFuture.resolve(InvocationFuture.java:101)
	at com.hazelcast.spi.impl.operationservice.impl.InvocationFuture.resolveAndThrow(InvocationFuture.java:75)
	at com.hazelcast.spi.impl.AbstractInvocationFuture.get(AbstractInvocationFuture.java:185)
	at com.hazelcast.util.FutureUtil.executeWithDeadline(FutureUtil.java:326)
	at com.hazelcast.util.FutureUtil.waitWithDeadline(FutureUtil.java:310)
	at com.hazelcast.util.FutureUtil.waitWithDeadline(FutureUtil.java:284)
	at com.hazelcast.transaction.impl.TransactionImpl.commit(TransactionImpl.java:292)
	at com.hazelcast.internal.cluster.impl.ClusterStateManager.changeClusterState(ClusterStateManager.java:277)
	at com.hazelcast.internal.cluster.impl.ClusterStateManager.changeClusterState(ClusterStateManager.java:244)
	at com.hazelcast.internal.cluster.impl.ClusterServiceImpl.changeClusterState(ClusterServiceImpl.java:854)
	at com.hazelcast.internal.cluster.impl.ClusterServiceImpl.changeClusterState(ClusterServiceImpl.java:849)
	at com.hazelcast.internal.cluster.impl.AdvancedClusterStateTest.changeClusterStateEventually(AdvancedClusterStateTest.java:806)
	at com.hazelcast.internal.partition.GracefulShutdownTest$6.run(GracefulShutdownTest.java:459)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:471)
	at java.util.concurrent.FutureTask.run(FutureTask.java:262)
	at java.lang.Thread.run(Thread.java:745)
```

https://hazelcast-l337.ci.cloudbees.com/view/Hazelcast/job/Hazelcast-3.x-OracleJDK1.7/com.hazelcast$hazelcast/651/testReport/junit/com.hazelcast.internal.partition/GracefulShutdownTest/shutdownMemberAndCluster_concurrently_withoutPartitionInitialization/

**Comment by hasancelik** [comment]
```
java.util.concurrent.ExecutionException: com.hazelcast.core.OperationTimeoutException: ChangeClusterStateOperation invocation failed to complete due to operation-heartbeat-timeout. Current time: 2016-11-09 05:37:57.806. Start time: 2016-11-09 05:35:57.555. Total elapsed time: 120251 ms. Last operation heartbeat: never. Last operation heartbeat from member: never. Invocation{op=com.hazelcast.internal.cluster.impl.operations.ChangeClusterStateOperation{serviceName='hz:core:clusterService', identityHash=1809387280, partitionId=-1, replicaIndex=0, callId=0, invocationTime=1478662557555 (2016-11-09 05:35:57.555), waitTimeout=-1, callTimeout=60000}, tryCount=250, tryPauseMillis=500, invokeCount=1, callTimeoutMillis=60000, firstInvocationTimeMs=1478662557555, firstInvocationTime='2016-11-09 05:35:57.555', lastHeartbeatMillis=0, lastHeartbeatTime='1970-01-01 02:00:00.000', target=[127.0.0.1]:12120, pendingResponse={VOID}, backupsAcksExpected=0, backupsAcksReceived=0, connection=MockConnection{localEndpoint=[127.0.0.1]:12123, remoteEndpoint=[127.0.0.1]:12120}}
	at java.util.concurrent.FutureTask.report(FutureTask.java:122)
	at java.util.concurrent.FutureTask.get(FutureTask.java:188)
	at com.hazelcast.internal.partition.GracefulShutdownTest.shutdownMemberAndCluster_concurrently(GracefulShutdownTest.java:465)
	at com.hazelcast.internal.partition.GracefulShutdownTest.shutdownMemberAndCluster_concurrently_withoutPartitionInitialization(GracefulShutdownTest.java:430)
Caused by: com.hazelcast.core.OperationTimeoutException: ChangeClusterStateOperation invocation failed to complete due to operation-heartbeat-timeout. Current time: 2016-11-09 05:37:57.806. Start time: 2016-11-09 05:35:57.555. Total elapsed time: 120251 ms. Last operation heartbeat: never. Last operation heartbeat from member: never. Invocation{op=com.hazelcast.internal.cluster.impl.operations.ChangeClusterStateOperation{serviceName='hz:core:clusterService', identityHash=1809387280, partitionId=-1, replicaIndex=0, callId=0, invocationTime=1478662557555 (2016-11-09 05:35:57.555), waitTimeout=-1, callTimeout=60000}, tryCount=250, tryPauseMillis=500, invokeCount=1, callTimeoutMillis=60000, firstInvocationTimeMs=1478662557555, firstInvocationTime='2016-11-09 05:35:57.555', lastHeartbeatMillis=0, lastHeartbeatTime='1970-01-01 02:00:00.000', target=[127.0.0.1]:12120, pendingResponse={VOID}, backupsAcksExpected=0, backupsAcksReceived=0, connection=MockConnection{localEndpoint=[127.0.0.1]:12123, remoteEndpoint=[127.0.0.1]:12120}}
	at com.hazelcast.spi.impl.operationservice.impl.InvocationFuture.newOperationTimeoutException(InvocationFuture.java:151)
	at com.hazelcast.spi.impl.operationservice.impl.InvocationFuture.resolve(InvocationFuture.java:101)
	at com.hazelcast.spi.impl.operationservice.impl.InvocationFuture.resolveAndThrow(InvocationFuture.java:75)
	at com.hazelcast.spi.impl.AbstractInvocationFuture.get(AbstractInvocationFuture.java:185)
	at com.hazelcast.util.FutureUtil.executeWithDeadline(FutureUtil.java:326)
	at com.hazelcast.util.FutureUtil.waitWithDeadline(FutureUtil.java:310)
	at com.hazelcast.util.FutureUtil.waitWithDeadline(FutureUtil.java:284)
	at com.hazelcast.transaction.impl.TransactionImpl.commit(TransactionImpl.java:292)
	at com.hazelcast.internal.cluster.impl.ClusterStateManager.changeClusterState(ClusterStateManager.java:277)
	at com.hazelcast.internal.cluster.impl.ClusterStateManager.changeClusterState(ClusterStateManager.java:244)
	at com.hazelcast.internal.cluster.impl.ClusterServiceImpl.changeClusterState(ClusterServiceImpl.java:854)
	at com.hazelcast.internal.cluster.impl.ClusterServiceImpl.changeClusterState(ClusterServiceImpl.java:849)
	at com.hazelcast.internal.cluster.impl.AdvancedClusterStateTest.changeClusterStateEventually(AdvancedClusterStateTest.java:806)
	at com.hazelcast.internal.partition.GracefulShutdownTest$6.run(GracefulShutdownTest.java:459)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:471)
	at java.util.concurrent.FutureTask.run(FutureTask.java:262)
	at java.lang.Thread.run(Thread.java:745)
```

https://hazelcast-l337.ci.cloudbees.com/view/Hazelcast/job/Hazelcast-3.x-OracleJDK1.7/com.hazelcast$hazelcast/664/testReport/junit/com.hazelcast.internal.partition/GracefulShutdownTest/shutdownMemberAndCluster_concurrently_withoutPartitionInitialization/

**Comment by ihsandemir** [comment]
Another failure here: https://hazelcast-l337.ci.cloudbees.com/job/new-lab-fast-pr/5887/testReport/junit/com.hazelcast.internal.partition/GracefulShutdownTest/shutdownAllMembers_withPartitionInitialization/

**Comment by metanet** [comment]
Fixed by #9324
